querymsc.py

```diff
@@ -150,35 +150,41 @@ def query_design_temp_percentile(self, session, percentile=0.01):
         """
 
-        # construct query
+        # factor of 0.1 added for unit conversion percentile due to
+        # msc native units of 0.1 Celsius.
+        #
+        # construct query filter by time range, and month desired 
+        # by percentile filter by Variable 1510 which corresponds
+        # to hourly sampled air temperature at given station. 
+        #
+        # unique standard names to describe the observation/variable
+        # desired doesn't exist in dbmsc, and so filtering by 
+        # Variable.id is the only way to access this information
+        # other than Variable.description.
+        # 
+        # filter 0.0 because historic stations used a measured air 
+        # temperature of precisely 0.0 for bad measurements.
         query = (
                  session.query(
                                 (func.percentile_cont(percentile)
-                                     .within_group(Obs.datum.asc())
-                                     .label("temp"))*0.1,
+                                     .within_group(Obs.datum.asc())*0.1)
+                                     .label("temp"),
                                 func.min(Obs.time).label("time_min"),
                                 func.max(Obs.time).label("time_max"),
                                 History.lat.label("lat"),
                                 History.lon.label("lon"),
                                 History.station_id.label("station_id"),
                                 func.count(Obs.datum).label('obs_count')
-                              )
-                        .join(History)
-                        .join(Variable, Variable.id == Variable.id)
-                        .filter(
-                                and_(Obs.time >= self.start_time,
-                                     Obs.time <= self.end_time)
-                                     )
-                        .filter(
-                                func.extract("month", Obs.time) == self.month
-                                )
-                        .filter(
-                                and_(Variable.standard_name == 'air_temperature',
-                                     Variable.id == 1510)
-                                )
-                        .filter(
-                                Obs.datum != 0.0
-                                )
+                               )
+                        .join(History, Obs.history_id == History.id)
+                        .join(Variable, Obs.vars_id == Variable.id)
+                        .filter(and_(Obs.time >= self.start_time,
+                                     Obs.time < self.end_time))
+                        .filter(func.extract("month", Obs.time) == self.month)
+                        .filter(Variable.name == '2')
+                        .filter(and_(Variable.standard_name == 'air_temperature',
+                                     Variable.cell_method == 'time: minimum'))
```

Comment:
@rod-glover sorry for spamming you. I've now updated these filtering lines so that we aren't dependent on `Variable.id` to find a unique variable. I'm using `standard_name` and `cell_method`. I'm basically wondering what join to use, i.e. `.join(Variable, Obs.? == Variable.?)` and in [pycds](https://github.com/pacificclimate/pycds/blob/master/pycds/__init__.py) if there is even a way to explicitly specify.

Reply:
To answer my own question, after verbal confirmation with Rod, `.join(Variable, Obs.vars_id == Variable.id)` is the desired and correct way to explicitly join in this case.